ci-operator/step-registry/cucushift/installer/check/reboot-nodes/cucushift-installer-check-reboot-nodes-commands.sh

```diff
@@ -4,10 +4,376 @@ set -o nounset
+    exit 1
+}
+
+# Add ERR trap to catch any unexpected exits
+trap 'error_handler $LINENO "$BASH_COMMAND"' ERR
+
+# Enhanced exit handler
+exit_handler() {
+    local exit_code=$?
+    log "=========================================="
+    log "Script execution completed"
+    log "=========================================="
+    log "Final exit code: ${exit_code}"
+    log "Execution time: $((SECONDS)) seconds"
+    log "Current time: $(get_timestamp)"
+    log "=========================================="
+}
+
+# Add EXIT trap to log script completion
+trap 'exit_handler' EXIT
+
+# Configuration constants - simplified for AWS hard reboot
+readonly AWS_INSTANCE_TIMEOUT=60    # 60 attempts × 10 seconds = 10 minutes
+readonly CLUSTER_RECOVERY_TIMEOUT=120  # 120 attempts × 30 seconds = 60 minutes
+
+# Reboot type configuration
+# HARD_REBOOT: Use cloud provider commands (AWS EC2 stop/start, etc.)
+# SOFT_REBOOT: Use SSH-based reboot (existing behavior)
+readonly REBOOT_TYPE="${REBOOT_TYPE:-SOFT_REBOOT}"
+
+# Record script start time for execution time calculation
+SECONDS=0
+
+log "=========================================="
+log "Script execution started"
+log "=========================================="
+log "Script: $0"
+log "Arguments: $*"
+log "Process ID: $$"
+log "Working directory: $(pwd)"
+log "User: $(whoami)"
+log "=========================================="
+
+# Platform detection and setup using case statement
+case "${CLUSTER_TYPE}" in
+    aws)
+        # Configure AWS credentials - support both CI and local environments
+        export AWS_REGION="${LEASED_RESOURCE}"
+        
+        # Determine AWS credentials file location
+        aws_cred_file=""
+        if [[ -n "${CLUSTER_PROFILE_DIR:-}" && -f "${CLUSTER_PROFILE_DIR}/.awscred" ]]; then
+            # CI environment
+            aws_cred_file="${CLUSTER_PROFILE_DIR}/.awscred"
+            export AWS_SHARED_CREDENTIALS_FILE="${aws_cred_file}"
+            log "Using CI environment AWS credentials: ${aws_cred_file}"
+        elif [[ -f "${HOME}/.aws/credentials" ]]; then
+            # Local environment
+            aws_cred_file="${HOME}/.aws/credentials"
+            export AWS_SHARED_CREDENTIALS_FILE="${aws_cred_file}"
+            log "Using local environment AWS credentials: ${aws_cred_file}"
+        else
+            log "ERROR: No AWS credentials file found"
+            log "Expected locations:"
+            log "  - CI environment: ${CLUSTER_PROFILE_DIR:-<unset>}/.awscred"
+            log "  - Local environment: ${HOME}/.aws/credentials"
+            exit 1
+        fi
+        
+        # Extract AWS credentials from the credential file (for CI environment)
+        if [[ "${aws_cred_file}" == *"/.awscred" ]]; then
+            AWS_ACCESS_KEY_ID=$(cat "${aws_cred_file}" | grep aws_access_key_id | tr -d ' ' | cut -d '=' -f 2)
+            AWS_SECRET_ACCESS_KEY=$(cat "${aws_cred_file}" | grep aws_secret_access_key | tr -d ' ' | cut -d '=' -f 2)
```

Comment:
The above `AWS_SHARED_CREDENTIALS_FILE` is satisfied to communicate with AWS services, we usually don't use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.